packages/zowe-explorer-api/src/profiles/ProfilesCache.ts

```diff
@@ -476,14 +476,14 @@ export class ProfilesCache {
         return parentProfile;
     }
 
-    private shouldRemoveTokenFromProfile(profile: imperative.IProfileLoaded, baseProfile: imperative.IProfileLoaded): boolean {
+    public shouldRemoveTokenFromProfile(profile: imperative.IProfileLoaded, baseProfile: imperative.IProfileLoaded): boolean {
         return ((baseProfile?.profile?.host || baseProfile?.profile?.port) &&
             profile?.profile?.host &&
             profile?.profile?.port &&
             (baseProfile?.profile.host !== profile?.profile.host ||
                 baseProfile?.profile.port !== profile?.profile.port ||
                 (profile?.profile.user && profile?.profile.password)) &&
-            profile?.profile.tokenType?.startsWith(imperative.SessConstants.TOKEN_TYPE_APIML)) as boolean;
+            baseProfile?.profile.tokenType?.startsWith(imperative.SessConstants.TOKEN_TYPE_APIML)) as boolean;
```

Comment:
Wouldn't this cause problems for extenders who provide different token types in their profile types?

For instance, if I have a profile which supports `JwtAuthentication` as my token type, this check right here may forcefully remove the token type from my extender-service profile if the default/given base profile contains an APIML tokenType.

Reply:
Sure enough, this change seems to impact the use of other tokens for z/OSMF (and may apply to extenders).

For example...
![image](https://github.com/user-attachments/assets/81d33992-dcce-4d49-a801-daf5e61dc17f)

This `zosmf_tokens.1` profile works on 3.1.2 and doesn't on this PR 😢